src/com/owncloud/android/datamodel/FileDataStorageManager.java

```diff
@@ -790,23 +867,45 @@ private boolean copyFile(File src, File target) {
                     new String[]{String.valueOf(parentId)}, null);
         }
 
-        if (c.moveToFirst()) {
-            do {
-                OCFile child = createFileInstance(c);
-                // TODO Enable when "On Device" is recovered ?
-                // if (child.isFolder() || !onlyOnDevice || onlyOnDevice && child.isDown()){
-                ret.add(child);
-                // }
-            } while (c.moveToNext());
+        if (c!= null) {
```

Comment:
No, this is not a good idea. We are already updating the property ```FILE_KEEP_IN_SYNC``` for all the descendents when its value is changed by the user for a folder.

For descendent files/folders that are created or discovered (from the server) after the user sets a folder as available offline, we should write the appropriate inherited value then - better increase time when writing the database than when reading it. 

So let's remove this from here and store ```AVAILABLE_OFFLINE_PARENT``` correctly instead of inferring it on reads.